### PR TITLE
DOC: many small fixes

### PR DESCRIPTION
**Pull request by elliotwaite**
Some suggestions for the docs.

If any of these suggestions are incorrect or unwanted, feel free to edit this PR as needed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ from jax import grad, jit, vmap
 def predict(params, inputs):
   for W, b in params:
     outputs = jnp.dot(inputs, W) + b
-    inputs = jnp.tanh(outputs)
+    outputs = jnp.tanh(outputs)
```
Comment (jakevdp) [inline README.md:66]: This code is correct as written (the outputs of one iteration become the inputs of the next iteration)

File: cloud_tpu_colabs/JAX_demo.ipynb [hunk 2]
```diff
@@ -252,7 +252,7 @@
     "def predict(params, inputs):\n",
     "  for W, b in params:\n",
     "    outputs = jnp.dot(inputs, W) + b\n",
-    "    inputs = jnp.tanh(outputs)\n",
+    "    outputs = jnp.tanh(outputs)\n",
```
Comment (jakevdp) [inline cloud_tpu_colabs/JAX_demo.ipynb:255]: Same here

**Review by jakevdp — CHANGES_REQUESTED** [review]
Thanks! Looks great, aside from two code examples that were correct as written.

File: README.md (hunk 1 above)
Comment (elliotwaite) [reply README.md:66]: Ah! I think my brain wasn't registering the for loop for some reason. Thanks for clarifying.

Comment (elliotwaite) [reply README.md]: This has been reverted in the latest commit.

File: cloud_tpu_colabs/JAX_demo.ipynb (hunk 2 above)
Comment (elliotwaite) [reply cloud_tpu_colabs/JAX_demo.ipynb]: This has been reverted in the latest commit.

**Comment by elliotwaite** [comment]
Thanks for the review. I've reverted those incorrect code changes.

**Comment by jakevdp** [comment]
Thanks - could you squash the changes into a single commit?

**Review by jakevdp — APPROVED** [review]

**Comment by jakevdp** [comment]
Also, we'll need to fix the CI failure (the ipynb and md versions of `Writing_custom_interpreters_in_Jax` do not match).

See https://jax.readthedocs.io/en/latest/developer.html#update-notebooks

**Comment by elliotwaite** [comment]
I fixed the ipynb/md issue.

Also, I think I squashed all the changes into the single commit titled "DOC: many small fixes", but this is my first time trying squashing so let me know if I did it incorrectly.

**Comment by jakevdp** [comment]
Thanks! I still see 7 commits on the branch. You can find some info about squashing e.g. here: https://stackoverflow.com/q/5189560

Once you squash the commits locally, you can force push your branch and it will be updated in the pull request. Thanks!

**Comment by jakevdp** [comment]
Since this touches so many files, it may also be a good idea to rebase on the main branch to pick up any conflicts that may have arisen since the initial commit.

**Comment by jakevdp** [comment]
Sorry about all of this – I know it's not a normal requirement for github PRs in other packages, but it's necessary due to the system we have in place to keep internal & external repositories in sync.

**Comment by jakevdp** [comment]
Thanks!

**Comment by elliotwaite** [comment]
I think I finally figured out the squash and rebase. The force push tip was the key, thanks. And thanks for being patient with my GitHub skills. This has been a good opportunity for me to learn more about how Git works.

**Comment by jakevdp** [comment]
Awesome! Thanks so much for the contribution 😁

**Pull request merged**